src/weatherSource/WeatherSource.js

```diff
@@ -0,0 +1,129 @@
+import {
+    relative
+} from "path";
+
+const API_KEY = '953349aaa2569f9cd4821f8c2ffda23a'
```

Comment:
Masz zmienną/stałą API_KEY (btw, średnik? ;) ). Ta zmienna nie jest niezależna od klasy, przy której została stworzona. To wyraźnie sugeruje, że albo powinna stanowić element klasy... chociaż tak naprawdę, to zamiast robić klasę i statyczne metody, powinno się **zrobić klasę, której obiekt się tworzy, podając odpowiedni API key w konstruktorze** (ten API_KEY może ewentualnie być wartością domyślną). Adres w zasadzie też by się przydało tak dostarczać.

W naszym projekcie, czysto edukacyjnym w sumie, to nie ma wielkiego znaczenia, ale to, co zrobiłeś, to jest bardzo zła praktyka. Wyobraź sobie, że OpenWeatherMap się przeprowadza, albo że serwis się rozrósł, jest faktycznie wykorzystywany i darmowy limit zapytań to za mało - więc trzeba wykupić płatną subskrypcję. A to może oznaczać nowy API Key. Co wtedy? W Twojej wersji oznacza to, że trzeba przekopywać się przez moduły i ręcznie przepisywać wartości na nowe. Inaczej się nie da, wszystko jest hard-coded. To jest olbrzymi tzw. dług technologiczny i o ile w naszym projekcie znaczenie ma ograniczone, o tyle "w szerokim świecie" tworzenia prawdziwych aplikacji, ludzie mogliby zapałać do Ciebie żądzą zemsty ;)

Reply:
Hmm, faktycznie ma to sens. Zmienię to. Na samym początku zastanawiałem się nad ideą statycznej klasy, bo bądź co bądź to wykorzystujemy to tylko do naszego projektu, który nie będzie później rozwijany, więc zostałem przy klasie statycznej. Jednak teraz widzę, że tworzenie obiektów może być bardziej przydatne, nawet jeśli chodzi o moją dobrą praktykę :P.

Co do tego API_key to z początku chciałem ustawić go na `const` w klasie jednak tak się nie dało, więc wyrzuciłem go poza klasę(co estetycznie mnie bardzo denerwowało)

Średnik ?  nie ma tam żadnego, chyba, że mówisz o `import relative from path`, który został wygenerowany automatycznie.

Reply:
JS nie pozwala deklarować stałych w środku klasy, też się z tym spotkałem, Gustaw polecał robić je poza klasą tak jak tu masz (i ja z resztą mam u siebie).

Reply:
Da się i nie da jednocześnie. Tak jak w Javie to nie da się zrobić. Ale JS ma specyficzne podejście do getterów. 

**Więc TO NIE ZADZIAŁA:**
```
class MyClass {
    static const MY_CONSTANT_VALUE = "hurra";
}

console.log(MyClass.MY_CONSTANT_VALUE);
```

**Ale można to zrobić tak i BĘDZIE DZIAŁAĆ:**
```
class MyClass {
    static get MY_CONSTANT_VALUE(){ return "hurra"; }
}

console.log(MyClass.MY_CONSTANT_VALUE);
```